git/operations.go

```diff
@@ -276,15 +282,37 @@ func changed(ctx context.Context, workingDir, ref string, subPaths []string) ([]
 	return splitList(out.String()), nil
 }
 
-func execGitCmd(ctx context.Context, args []string, config gitCmdConfig) error {
-	if trace {
-		print("TRACE: git")
-		for _, arg := range args {
-			print(` "`, arg, `"`)
+// traceGitCommand returns a log line that can be useful when debugging and developing git activity
+func traceGitCommand(args []string, config gitCmdConfig, stdout string, stderr string) string {
```

Comment:
BTW you can collapse adjacent variables or arguments of the same type: `..., stdout, stderr string`

Reply:
Although the standard library and the Go source code does that a lot I tend to not opt to because if you ever need to refactor something I find it a problem that if you have: a signature like `func example(firstArg, secondArg string)` that if you should want to change `secondArg` to something else that you would need to make a change to firstArg.  This happened to me a lot,[ for example](https://github.com/dimitropoulos/flux/commit/d3d6248d1da783a1df76b8fca11063060282d31d), when I was refactoring the usages of git refs to better understand how they're used in the application.  Although that was mostly for my own understanding it was extremely tedious to have to constantly undo this "shortcut".

Also I find it unnecessarily complicated to not be able to see what the type of something without reading far down the line (e.g. `func clone(ctx context.Context, workingDir, repoURL, repoBranch string)`).  My attitude is that source code should be optimized for reading not writing since reading code is where the vast majority of time is spent - that's why I don't do it.

I'll keep that in mind in the future.

**Would you like me to make that change before merging?**  I only ask because you made this comment after approving.

Reply:
> Would you like me to make that change before merging?

Nahp, no need to change it.